File: src/python/huiwen.py
# coding=utf-8
""" 回文数 """

import datetime
from numba import jit

"""
是否回文
"""
def ishuiwen(n:int):
    sn = 0 # 可以不用类型
    sn = n
    tn = 0
    while sn != 0:
        tn = tn * 10 + sn % 10
        sn = sn // 10
    if tn == n:
        return True
    return False

"""
回文10000000
"""
def huiwen(max:int=10000000):
    tx = 0
    for i in range(0, max):
        # print(f"数:{i}")
        if ishuiwen(i):
            tx+=1
            # print(f"回文数:{i}")
    return tx

@jit
def run():
    t1 = datetime.datetime.now()
    r = huiwen()
    t2 = datetime.datetime.now()
    tx = t2 - t1
    print(f"回文 结果:: {r} == 时间::[{tx.seconds}]秒{tx.microseconds/1000}毫秒")


if __name__ == '__main__':
    run()
# python .\huiwen.py;python --version

File: src/julia/huiwen.jl
# 20200816
using Dates
using Printf

"""
是否回文
"""
function ishuiwen(n::Int64)
    sn::Int32 = 0 # 可以不用类型
    sn = n
    tn::Int32 = 0
    while sn != 0
        tn = tn * 10 + sn % 10
        sn = sn ÷ 10
    end
    if tn == n
        return true
    end
    return false
end

"""
回文10000000
"""
function huiwen(max::Int64=10000000)
    tx::Int32 = 0
    x::Int32 = 0
    for i = 0:max
        if ishuiwen(i) == true
            tx+=1
        end
    end
    return tx
end

function run()
    t1 = now()
    r = huiwen(10000000)
    t2 = now()
    tx = t2 - t1
    @printf("回文 结果:: %d == 时间:: %F \n", r, Dates.value(tx))
end

run()
# julia .\huiwen.jl; julia --version